### PR TITLE
refactor: Message

### PR DESCRIPTION
**Pull request by pajlada**
# Description

this PR was cherry-picked into smaller PRs

 - [x] Virtual-override-related changes: https://github.com/Chatterino/chatterino2/pull/4917
 - [x] FilterParser-changes: https://github.com/Chatterino/chatterino2/pull/4925
 - [x] ChannelView changes: https://github.com/Chatterino/chatterino2/pull/4926
 - [x] IrcMessageHandler changes: https://github.com/Chatterino/chatterino2/pull/4927

With the last remaining changes to `Message` & some other small changes staying here

Message: Remove empty anon namespace
Message: Remove else after return
Message: Avoid repeating type in return
Message: Remove unused includes (there were a few)
Helix: Remove unneccesary `static` for already-static variables
AttachedWindow: Remove unused ForwardDecl.hpp include

### CODE REVIEW

**Comment by Mm2PL** [comment]
image
lgtm

File: src/messages/Message.cpp [hunk 1]
```diff
@@ -26,45 +19,57 @@ Message::~Message()
...
+    if (this->flags.has(MessageFlag::FirstMessage))
     {
-        return SBHighlight(
+        return {
             ColorProvider::instance().color(ColorType::FirstMessageHighlight),
-            SBHighlight::Default, false, true);
+            ScrollbarHighlight::Default,
+            false,
+            true,
+        };
```
Comment (Nerixyz) [inline src/messages/Message.cpp:69]: Not from this PR: The `ScrollbarHighlight` needs some refactoring. The arguments `isRedeemedHighlight`,
`isFirstMessageHighlight`, and `isElevatedMessageHighlight` are mutually exclusive - shouldn't this be refactored to an enum? Even if they're not mutually exclusive, a bit field would be more readable.

**Review by Nerixyz — COMMENTED** [review]
For a future refactoring: There are a few `#pragma comment(lib, "my.lib")` in the code (e.g. in _AttachedWindow.cpp_). These can be replaced by `target_link_libraries(<target> PRIVATE my.lib)` (maybe there's even a better way of doing this in CMake).

File: src/messages/Message.cpp (hunk 1 above)
Comment (pajlada) [reply src/messages/Message.cpp:69]: Definitely should be enums yeah

**Review by pajlada — COMMENTED** [review]
image

**Pull request merged**